src/AWS/Orleans.Persistence.DynamoDB/Provider/DynamoDBGrainStorage.cs

```diff
@@ -185,7 +185,7 @@ public async Task WriteStateAsync<T>(string grainType, GrainReference grainRefer
             }
             catch (ConditionalCheckFailedException exc)
             {
-                throw new InconsistentStateException("Invalid grain state", exc);
+                throw new InconsistentStateException($"Inconsistent grain state: {exc}");
```

Comment:
Does this better align with other providers, or is there some other reasoning for this? Mostly just curious.

Reply:
Yeah, that would bite us with TypeNotFoundException. Good catch.